fmriprep/workflows/base.py

```diff
@@ -491,6 +491,10 @@ def init_single_subject_wf(
     else:
         subject_data = collect_data(layout, subject_id, task_id, echo_idx,
                                     bids_filters=bids_filters)[0]
+        if 'flair' in ignore and subject_data['flair']:
+            subject_data['flair'] = []
+        if 't2w' in ignore and subject_data['t2w']:
```

Comment:
No harm in replacing empty lists.

```suggestion
        if 'flair' in ignore:
            subject_data['flair'] = []
        if 't2w' in ignore:
```